ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/TransactionSimulator.java

```diff
@@ -141,14 +145,77 @@ public Optional<TransactionSimulatorResult> process(
         blockHeader);
   }
 
+  public Optional<TransactionSimulatorResult> processOnPending(
+      final CallParameter callParams,
+      final Optional<AccountOverrideMap> maybeStateOverrides,
+      final TransactionValidationParams transactionValidationParams,
+      final OperationTracer operationTracer,
+      final ProcessableBlockHeader pendingBlockHeader) {
+
+    try (final MutableWorldState disposableWorldState =
+        duplicateWorldStateAtParent(pendingBlockHeader.getParentHash())) {
+      WorldUpdater updater = getEffectiveWorldStateUpdater(disposableWorldState);
+
+      // in order to trace the state diff we need to make sure that
+      // the world updater always has a parent
+      if (operationTracer instanceof DebugOperationTracer) {
+        updater = updater.parentUpdater().isPresent() ? updater : updater.updater();
+      }
+
+      return processWithWorldUpdater(
+          callParams,
+          maybeStateOverrides,
+          transactionValidationParams,
+          operationTracer,
+          pendingBlockHeader,
+          updater,
+          Address.ZERO);
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  public ProcessableBlockHeader simulatePendingBlockHeader() {
```

Comment:
would it be useful to have a debug or trace log for when this is called

Reply:
done